v3/test/helpers.go

```diff
@@ -35,7 +39,15 @@ import (
 // lint result is nil.
 //nolint:revive
 func TestLint(lintName string, testCertFilename string) *lint.LintResult {
-	return TestLintCert(lintName, ReadTestCert(testCertFilename))
+	return TestLintWithConfig(lintName, testCertFilename, "")
+}
+
+func TestLintWithConfig(lintName string, testCertFilename string, context string) *lint.LintResult {
+	ctx, err := lint.NewConfigFromString(context)
```

Comment:
nit: calling it a `context` and naming the variable `ctx` is likely to be confusing to people who tightly associate that name with `context.Context`. I'd rename these `configStr` and `config`, or `config` and `cfg`, or something like that.

The same also applies a couple times below where the name `ctx` is used for a `lint.Configuration`.